packages/cache-core/src/strategies/fileSystem.ts

```diff
@@ -1,21 +1,30 @@
 import fs from 'node:fs/promises'
+import { existsSync } from 'node:fs'
 import path from 'node:path'
 import type { CacheStrategy, CacheEntry, CacheContext } from '@dbbs/next-cache-handler-common'
 
 export class FileSystemCache implements CacheStrategy {
   async get(pageKey: string, cacheKey: string, ctx: CacheContext): Promise<CacheEntry | null> {
-    const data = await fs.readFile(path.join(ctx.serverCacheDirPath, pageKey, `${cacheKey}.json`), {
-      encoding: 'utf-8'
-    })
+    const pathToCacheFile = path.join(ctx.serverCacheDirPath, pageKey, `${cacheKey}.json`)
+
+    if (!existsSync(pathToCacheFile)) {
+      throw `no cache file ${pathToCacheFile} yet`
+    }
+
+    const data = await fs.readFile(pathToCacheFile, 'utf-8')
 
     if (!data) return null
 
     return JSON.parse(data)
   }
 
   async set(pageKey: string, cacheKey: string, data: CacheEntry, ctx: CacheContext): Promise<void> {
-    const filePath = path.join(ctx.serverCacheDirPath, pageKey, `${cacheKey}.json`)
-    await fs.writeFile(filePath, JSON.stringify(data))
+    const pathToCacheFolder = path.join(ctx.serverCacheDirPath, pageKey)
+    const pathToCacheFile = path.join(pathToCacheFolder, `${cacheKey}.json`)
+
+    if (!existsSync(pathToCacheFolder)) await fs.mkdir(pathToCacheFolder, { recursive: true })
```

Comment:
let's avoid doing that each time when we call `set`, better to make it in constructor of class, so we should run it once only

Reply:
I thought about it, but it looks like we don't know all possible page keys in the constructor. 
Am I right?

Reply:
Can we change the structure of cache files (avoid nesting and make them flat)? Or we can add catch for this case and make a folder only after a file write error, and then write successfully after that

Reply:
we do not need to know page keys, because that method creates them, we just need to create cache folder where we should write keys

Reply:
so we can create folder once when initialize class